server/main.go

```diff
@@ -0,0 +1,79 @@
+package main
+
+import (
+	"bufio"
+	"context"
+	"log"
+	"net"
+	"os"
+	pb "server/file_manager"
+
+	"google.golang.org/grpc"
+)
+
+type server struct {
+	pb.UnimplementedFileManagerServer
+}
+
+func (s *server) PutFile(ctx context.Context, in *pb.PutFileRequest) (*pb.PutFileResponse, error) {
+	var fileName = in.GetFileName()
+	var fileContext = in.GetFileContext()
+
+	log.Printf("Received fileName and fileContext as : %v", fileName+" | "+fileContext)
+
+	writeFile(fileName, fileContext)
+
+	// writeFile 에서 string을 리턴하고, 현재 함수에서 return값을 받아와서 PutFileResponse로 리턴할 수 없으면 결과 어떻게 로그찍음??
+	return &pb.PutFileResponse{Result: "File saved successfully"}, nil
+}
+
+func writeFile(fileName string, fileContext string) {
```

Comment:
```suggestion
func writeFile(fileName string, fileContext string) error {
```
Most functions should return an error, so that callers can handle it.

Reply:
Copy